### PR TITLE
Zhouyuan/optimize the marching component

### PR DESCRIPTION
**Pull request by daniel-zint**

### CODE REVIEW

**Comment by Zhouyuan-Chen** [comment]
had added the position update, but still have a problem with it.

Also, as I have a closer look at our current code, I don't think it is practical to do so since we may need some modification for more files to fulfill this component's demand.

**Comment by Zhouyuan-Chen** [comment]
I realized there is a bug in the marching component. I will fix it tomorrow.

File: components/tests/test_component_marching.cpp [hunk 1]
```diff
@@ -144,6 +152,21 @@ TEST_CASE("marching_component_tri", "[components][marching]")
         CHECK(isosurface_vertex_num == expected_isosurface_vertex_num);
     }
 
+    const auto& edges = m.get_all(PrimitiveType::Edge);
+    wmtk::attribute::MeshAttributeHandle edge_tag_handle =
+        m.get_attribute_handle<int64_t>("marching_edge_tag", PrimitiveType::Edge);
```
Comment (daniel-zint) [inline components/tests/test_component_marching.cpp:157]: The names for the edge and face tag attributes should be given by the user.

File: components/tests/test_component_marching.cpp [hunk 2]
```diff
@@ -167,25 +190,32 @@ TEST_CASE("marching_component_tri", "[components][marching]")
         }
     }
 
-    if (false) {
-        wmtk::io::ParaviewWriter
-            writer("marching_2d_result", "vertices", m, true, false, true, false);
+    if (true) {
+        wmtk::io::ParaviewWriter writer(
+            "/home/zhouyuan/workplace/toolkit/wildmeshing-toolkit/data/marching_2d_result",
```
Comment (daniel-zint) [inline components/tests/test_component_marching.cpp:195]: No absolute paths please.
Also this should not print anything by default --> `if(false)`

File: components/wmtk_components/marching/wmtk/components/marching/internal/Marching.cpp [hunk 3]
```diff
@@ -83,18 +114,66 @@ void Marching::process()
...
+
+        op_split.add_transfer_strategy(etag_strategy);
+
+        /**************************face tag******************************/
+        auto compute_face_label =
+            [this](const Eigen::MatrixX<int64_t>& labels) -> Eigen::VectorX<int64_t> {
+            assert(labels.cols() == 3);
+            if (labels(0, 0) == m_output_value && labels(0, 1) == m_output_value &&
+                labels(0, 2) == m_output_value)
+                return Eigen::VectorX<int64_t>::Constant(1, m_output_value);
```
Comment (daniel-zint) [inline components/wmtk_components/marching/wmtk/components/marching/internal/Marching.cpp:158]: add brackets around the return statement

**Review by daniel-zint — COMMENTED** [review]
Some minor comments. Please add the attributes to the json options. Otherwise I think it's fine.

**Pull request merged**